### DISCUSSION TITLE
markitdown for node

### DISCUSSION THREAD

**Discussion by jojomondag** [microsoft/markitdown#190] [Ideas]
Thank you for you project i took inspiration from it to my own project.

https://github.com/[jojomondag/FileToMarkdown](https://github.com/jojomondag/FileToMarkdown)

i have a npm package ready to download and use in your javascript code here.

https://www.npmjs.com/package/filetomarkdown

Maybe somebody needs a version working on their page you can use mine.

**Comment by jparkerweb** [comment]
I was doing something similar myself :)

[npm : down-craft](https://www.npmjs.com/package/down-craft)
[github : down-craft](https://github.com/jparkerweb/down-craft)

![image](https://github.com/user-attachments/assets/76db70fe-b596-4c39-9f9b-199167b4b639)

**Reply by jojomondag** [reply]
Cool love your image :P

**Comment by xeoshow** [comment]
Hi @jojomondag ,

Just would like to know: what version of markitdown is used in the filetomarkdown? Or is it a totally rewritten in nodejs? 

And: is there any nodejs API for filetomarkdown? I just see the restful API, but in our context we would like to use the nodejs API directly.

Thanks v much.

**Reply by jojomondag** [reply]
FileToMarkdown is completely written in Node.js i dont use markitdown its my own implementation. Some of my own classes are called markitdown that can lead to confusion thanks.

Maybe these Node.js APIs can work for you

const { MarkitDown, convertToMarkdown } = require('filetomarkdown');

// Simple conversion
const markdown = await convertToMarkdown('document.pdf', 'output.md');

// Class-based approach  
const converter = new MarkitDown();
const result = await converter.convertToMarkdown('input.docx');

Dont be afraid to do your own pullrequest with what you need.

Greetings/Jojo

**Reply by xeoshow** [reply]
Thanks for reply! 
Compared to markitdown, are all the features implemented in FileToMarkdown? Or is there any feature still missing?
Thanks again.

Best regards, Jason

**Reply by jojomondag** [reply]
No markitdown is more feature rich. I dont support sound and video.

**Reply by jojomondag** [reply]
And markitdown support much more formats.

**Comment by Diluka** [comment]
1. https://www.npmjs.com/package/markitdown-ts ts implementation 
2. https://www.npmjs.com/package/@mote-software/markitdown binary wrapper

**Reply by jojomondag** [reply]
Looks good! thanks for the find.

**Comment by clauxel** [comment]
I found this useful because it goes beyond “convert a file” and gets into whether the result is actually clean enough for a real workflow. If others here are looking for the official MarkItDown Online page, it is available at https://markitdown.store/
